examples/plot_survival_analysis.py

```diff
@@ -195,11 +195,74 @@
 ax.set_ylabel("objective suboptimality")
 _ = ax.set_xlabel("time in seconds")
 
-
-
 # %%
 # According to printed ratio, using ``skglm`` we get the same result as ``lifelines``
 # with more than x100 less time!
 speed_up = records["lifelines"]["times"][-1] / records["skglm"]["times"][-1]
 print(f"speed up ratio: {speed_up:.0f}")
 
+# %%
+# Efron estimate
+# ==============
+#
+# The previous results, namely closeness and timings, can be extended to the case
+# of handling tied observation with the Efron estimate.
+#
+# Let's start by generating data with tied observation. This can be achieved
+# by passing in a ``with_ties=True`` to ``make_dummy_survival_data`` function.
+tm, s, X = make_dummy_survival_data(
+    n_samples, n_features,
+    normalize=True,
+    with_ties=True,
+    random_state=0
+)
+
+# check data has tied observations
+print("Number of unique times", np.unique(tm))
+print("Number of samples", n_samples)
+
+# %%
+# It is straightforward to fit an :math:`\ell_1` Cox estimator with the Efron estimate.
+# We only need to pass in ``use_efron=True`` to the ``Cox`` datafit.
+
+# ensure using Efron estimate
```

Comment:
this is potentially dangerous in other cases (changing the parameter sometimes does not change it in an estimator). For pedagogical purposes it may be better to just reinstantiate a Cox datafit, what do you think ?

Reply:
It would be problematic if we change params and forget to call ``initialize``

Agree, we better avoid that. Also considering that we use ``lru_cache``, re-defining the datafit won't incur an overhead.

Open question: maybe we should think about a safe mechanism to prevent changing attributes in datafits?

Reply:
yes, the possibility to call solvers on datafits that have been initialized on other data is potentially harmful... and the need for "initialize" is not obvious to the average user.

Random thoughts:

I'm not sure the problem is solvable, for example in sklearn you can use a Lasso with warm_start=True, and if you fit a second time on an X with a different shape it'll break : 
![image](https://github.com/scikit-learn-contrib/skglm/assets/8993218/4988d6b8-f3d6-4f77-a4bb-3545544ef257)



In our situation I'm foreseeing cases with silent, hard to catch convergence issues because someone fits on another X with compatible shape but without recalling `initialize`.
If people use the `GeneralizedLinearEstimator` this won't happen because we call initialize inside.

we could gather all this in an issue for future discussion

Reply:
Open an issue https://github.com/scikit-learn-contrib/skglm/issues/164 to address that